src/Core/Domain/Shipment/QueryResult/OrderShipmentProduct.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Shipment\QueryResult;
+
+class OrderShipmentProduct
+{
+    /**
+     * @var int
+     */
+    private int $orderDetailId;
+
+    /**
+     * @var int
+     */
+    private int $quantity;
+
+    /**
+     * @var string
+     */
+    private ?string $productName;
+
+    public function __construct(
+        int $orderDetailId,
+        int $quantity,
+        ?string $productName
+    ) {
+        $this->orderDetailId = $orderDetailId;
+        $this->quantity = $quantity;
+        $this->productName = $productName;
+    }
```

Comment:
Little tip in case you don't know about it yet, since we are now using PHP 8.1 minimum we can use the property promotion in constructor, this reduces the size of the code and its readability (IMHO)

```suggestion
    public function __construct(
        private int $orderDetailId,
        private int $quantity,
        private ?string $productName
    ) {
    }
```

Reply:
About `productName`, can this be null? The getter method doesn't seem to translate this as it's always returning a string
I think you can force it be always be a string but empty value could be acceptable